packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -75,12 +82,13 @@ export type TabsProps = {
    * Content inside the tab panel.
    */
   children?: React.ReactNode;
-} & TabsOrientationProps;
+} & TabsTypeProps;
 
 /**
  * @deprecated Since v2, use `TabProps` with `Tabs`
  */
-export type HorizontalTabsProps = Omit<TabsProps, 'orientation'>;
+export type HorizontalTabsProps = Omit<TabsProps, 'orientation'> &
+  TabsTypeProps;
```

Comment:
I'm not sure if we need to add this, since `HorizontalTabsProps` is deprecated. But I had to add this to avoid the TS error on [L358](https://github.com/iTwin/iTwinUI/blob/20b2973655bd87b125bbd220dd9ebb60d0e1629d/packages/itwinui-react/src/core/Tabs/Tabs.tsx#L358)

![image](https://user-images.githubusercontent.com/45748283/212418024-b7da42ca-70f7-4b2a-b335-18c81ca2d70a.png)

Reply:
Removing `actions?: undefined;` in L41 should get rid of the error without adding TabsTypeProps one more time. It's already included in TabsProps.

Reply:
This doesn't make sense. We are omitting "orientation" prop and adding it back with TabsTypeProps. 
Yes, it does get rid of the typescript error but it creates different kind of problem.
Users can override "orientation" prop with vertical: 
<img width="558" alt="image" src="https://user-images.githubusercontent.com/83585998/214055879-d51d4526-3eb7-480a-b893-7f3a2439d057.png">

With current typings typescript thinks that everything with "horizontal" orientation needs to have type "pill". Maybe you can rethink TabsTypeProps?

Reply:
Not defining `actions: undefined` seemed to work. So now no change need to be done on the `HorizontalTabsProps` type. (https://github.com/iTwin/iTwinUI/pull/1002/files/55f64b16bcc8c5bc01ce2823f9f65f49a0ba6185..56b1c540a778ae26c1c327f53986c850c5e45a98)

But since `actions` are not existing when `type: pill`, we cannot get `actions` when destructuring `props`. Instead we have to access `actions` using `props.actions` and also check whether `props.type !== 'pill'` before accessing `props.actions` ([L339](https://github.com/iTwin/iTwinUI/pull/1002/files/55f64b16bcc8c5bc01ce2823f9f65f49a0ba6185..56b1c540a778ae26c1c327f53986c850c5e45a98#diff-b7bdf1756f6685cb3e8678eaedca9cac8289f5e72de3bb3381eb68bd7c31517aR339))

Just for clarity and to avoid logical type-related bugs, I split `TabsTypeProps` into `TabsOrientationProps` (from before this PR) and `TabsTypeProps` (new).